core/src/main/java/org/javarosa/core/model/SelectChoice.java

```diff
@@ -121,4 +121,46 @@ public String getTextID() {
     public void setTextID(String textID) {
         this.textID = textID;
     }
+
+    @Override
+    public int hashCode() {
+        int result;
+        result = textID != null ? textID.hashCode() : 0;
+        result = 31 * result + value.hashCode();
```

Comment:
ah, right. I was paying attention to when it is set via `ExtUtil.nullIfEmpty` 
https://github.com/dimagi/javarosa/blob/master/core/src/main/java/org/javarosa/core/model/SelectChoice.java#L93